src/builder/application-registration-builder.ts

```diff
@@ -0,0 +1,137 @@
+import { randomUUID } from "crypto";
+import { AuthorizationAgent } from "../authorization-agent";
+import {
+    AccessAuthorization,
+    AccessGrant,
+    Agent,
+    AgentRegistration,
+    ApplicationRegistration,
+    DataAuthorization,
+    DataGrant,
+    DataRegistration,
+    IDataGrantBuilder,
+    RdfFactory,
+    SocialAgent,
+} from "solid-interoperability";
+import { AuthorizationBuilder } from "./authorization-builder";
+import { insertTurtleResource, updateContainerResource } from "../utils/modify-pod";
+import { Session } from "@inrupt/solid-client-authn-node";
+import { parseTurtle } from "../utils/turtle-parser";
+
+export class AgentRegistrationBuilder {
+    private data_grants: DataGrant[] = [];
+    private access_grants: AccessGrant[] = [];
+    private registration: AgentRegistration | undefined = undefined;
+
+    constructor(private authorization_agent: AuthorizationAgent) {}
+
+    async build(registeredAgent: Agent, authorization_builder: AuthorizationBuilder[]) {
+        const id = this.authorization_agent.AgentRegistry_container + randomUUID() + "/";
+
+        let builders: GrantNeedGroupBuilder[] = [];
+        for (const builder of authorization_builder) {
+            const grant_builder = new GrantNeedGroupBuilder(this.authorization_agent, id);
+            await grant_builder.createDataGrants(builder.getCreatedDataAuthorizations());
+            await grant_builder.createAccessGrant(builder.getCreatedAccessAuthorization());
+
+            for (const grant of grant_builder.getDataGrants()) {
+                this.data_grants.push(grant);
+            }
+            this.access_grants.push(grant_builder.getAccessGrant());
+            builders.push(grant_builder);
+        }
+
+        this.registration = new ApplicationRegistration(
+            id,
+            this.authorization_agent.social_agent,
+            this.authorization_agent.authorization_agent,
+            new Date(),
+            new Date(),
+            registeredAgent,
+            builders.map((builder) => builder.getAccessGrant()),
+        );
+    }
+
+    async storeToPod() {
+        if (!this.registration)
+            throw new Error("Not able to store in pod, because AgentRegistration is not created");
```

Comment:
This will crash the service, but I'm guessing we cant reach this point without making an AgentRegistration?